### PR TITLE
Port last integration tests to python

### PR DESCRIPTION
**Pull request by hubertdeng123**

### CODE REVIEW

File: sentry-admin.sh [hunk 1]
```diff
@@ -19,7 +19,8 @@ on the host filesystem. Commands that write files should write them to the '/sen
 
 # Actual invocation that runs the command in the container.
 invocation() {
-  $dc run -v "$VOLUME_MAPPING" --rm -T -e SENTRY_LOG_LEVEL=CRITICAL web "$@"
+  output=$($dc run -v "$VOLUME_MAPPING" --rm -T -e SENTRY_LOG_LEVEL=CRITICAL web "$@" 2>&1)
```
Comment (hubertdeng123) [inline sentry-admin.sh:22]: This is a fix, I don't think this was properly working before

File: _integration-test/test_run.py [hunk 2]
```diff
@@ -138,24 +146,216 @@ def test_cleanup_crons_running():
 
 
 def test_custom_cas():
```
Comment (azaslavsky) [inline _integration-test/test_run.py:148]:
```suggestion
def test_custom_certificate_authorities():
```

nit: clearer for the uninitiated, imo

File: _integration-test/test_run.py [hunk 3]
```diff
@@ -138,24 +146,216 @@ def test_cleanup_crons_running():
...
+    # Generate tightly constrained CA
+    ca_key = rsa.generate_private_key(
+        public_exponent=65537, key_size=2048, backend=default_backend()
+    )
+
+    ca_name = x509.Name(
+        [x509.NameAttribute(NameOID.COMMON_NAME, "TEST CA *DO NOT TRUST*")]
+    )
+
+    ca_cert = (
```
Comment (azaslavsky) [inline _integration-test/test_run.py:166]: This test is about 1000x clearer than the one it was replacing, nicely done!

File: _integration-test/test_run.py [hunk 4]
```diff
@@ -138,24 +146,216 @@ def test_cleanup_crons_running():
...
+        ],
+        check=True,
+    )
+
+    # Remove files
+    os.remove(f"{custom_certs_path}/test-custom-ca-roots.crt")
+    os.remove("sentry/test-custom-ca-roots.py")
+
+    # Unset environment variable
+    if "COMPOSE_FILE" in os.environ:
```
Comment (azaslavsky) [inline _integration-test/test_run.py:357]: One thing I noticed: this test doesn't seem to have any asserts in it? Is there anything we can do to check that our custom CAs are actually the ones used?

File: sentry-admin.sh (hunk 1 above)
Comment (azaslavsky) [reply sentry-admin.sh:22]: 👍

File: _integration-test/test_backup.py [hunk 5]
```diff
@@ -1,25 +1,69 @@
-import subprocess
 import os
+import subprocess
+
+
+def test_sentry_admin(setup_backup_restore_env_variables):
```
Comment (azaslavsky) [inline _integration-test/test_backup.py:5]: Thanks for porting this! :)

File: _integration-test/test_run.py (hunk 4 above)
Comment (hubertdeng123) [reply _integration-test/test_run.py:357]: It actually does have asserts in it, although it's not very clear. I can put in a comment explaining this in the code, but the asserts come from `_integration-test/custom-ca-roots/test.py`. The file is copied into the docker volume and executed within the container with the command 

```
subprocess.run(
        [
            "docker",
            "compose",
            "--ansi",
            "never",
            "run",
            "--no-deps",
            "web",
            "python3",
            "/etc/sentry/test-custom-ca-roots.py",
        ],
        check=True,
    )
```

File: _integration-test/test_run.py (hunk 2 above)
Comment (hubertdeng123) [reply _integration-test/test_run.py]: done

**Review by azaslavsky — APPROVED** [review]

**Pull request merged**